src/main/java/org/testng/TestNG.java

```diff
@@ -269,28 +269,33 @@ public void setXmlPathInJar(String xmlPathInJar) {
   public void initializeSuitesAndJarFile() {
     // The Eclipse plug-in (RemoteTestNG) might have invoked this method already
     // so don't initialize suites twice.
+	System.out.println("ENTER");
     if (m_isInitialized) {
       return;
     }
 
     m_isInitialized = true;
     if (m_suites.size() > 0) {
-    	//to parse the suite files (<suite-file>), if any
-    	for (XmlSuite s: m_suites) {
-        for (String suiteFile : s.getSuiteFiles()) {
-            Path rootPath = Paths.get(s.getFileName()).getParent();
-            try {
-                Collection<XmlSuite> childSuites = getParser(rootPath.resolve(suiteFile).normalize().toString()).parse();
-                for (XmlSuite cSuite : childSuites){
-                    cSuite.setParentSuite(s);
-                    s.getChildSuites().add(cSuite);
-                }
+    //to parse the suite files (<suite-file>), if any
+	for (XmlSuite s: m_suites) {
+	   for (String suiteFile : s.getSuiteFiles()) {
+        	Path suitePath = Paths.get(s.getFileName());
+        	Path rootPath = suitePath.getParent();
+        	try {
+        		Collection<XmlSuite> childSuites = getParser(rootPath.resolve(suiteFile).normalize().toString()).parse();
+        		for (XmlSuite cSuite : childSuites){
+        			Path childSuite = Paths.get(cSuite.getFileName()).normalize();
+        			Path parentSuite = Paths.get(suiteFile).normalize();
```

Comment:
Paths.get(suiteFile).normalize() will get it from the project dir and rootPath.resolve(suiteFile).normalize() will first append the suiteFIle path and the normalize will remove the extra path identifiers. For ex, some users may write .\testng.xml and some may write \\tesng.xml.